packages/devextreme/js/__internal/ui/chat/messagelist.ts

```diff
@@ -169,6 +176,51 @@ class MessageList extends Widget<Properties> {
     });
   }
 
+  _shouldAddDateHeader(timestamp: undefined | string | number | Date): boolean {
+    const { showDateHeaders } = this.option();
+    if (timestamp === undefined || !showDateHeaders) {
+      return false;
+    }
+
+    const deserializedDate = dateSerialization.deserializeDate(timestamp);
+
+    if (isNaN(deserializedDate.getTime())) {
+      return false;
+    }
+
+    return !dateUtils.sameDate(this._messageDate, deserializedDate);
+  }
+
+  _createMessageDateHeader(timestamp: string | number | Date | undefined): void {
+    if (timestamp === undefined) {
```

Comment:
This check is excess while we call this method only when `_shouldAddDateHeader` returns true

Reply:
fair point, removed this